miner/worker.go

```diff
@@ -991,10 +985,30 @@ func (w *worker) commit(uncles []*types.Header, interval func(), update bool, st
 	return nil
 }
 
+// deep copy current receipts. This is should be used whene an asynchronus process can modify
+// current context.
+func (w *worker) copyCurrentReceipts() []*types.Receipt {
+	receipts := make([]*types.Receipt, len(w.current.receipts))
+	for i, l := range w.current.receipts {
+		cpy := *l
+		receipts[i] = &cpy
+	}
+	return receipts
+}
+
 // postSideBlock fires a side chain event, only use it for testing.
 func (w *worker) postSideBlock(event core.ChainSideEvent) {
 	select {
 	case w.chainSideCh <- event:
 	case <-w.exitCh:
 	}
 }
+
+// computes total consumed fees in ETH. Block transactions and receipts have to have the same order.
+func totalFees(block *types.Block, receipts []*types.Receipt) *big.Float {
+	feesWei := new(big.Int)
+	for i, tx := range block.Transactions() {
+		feesWei.Add(feesWei, new(big.Int).Mul(new(big.Int).SetUint64(receipts[i].GasUsed), tx.GasPrice()))
+	}
+	return new(big.Float).Quo(new(big.Float).SetInt(feesWei), new(big.Float).SetInt(big.NewInt(params.Ether)))
+}
```

Comment:
This method is again a small utility code that's used solely for 1 log line. Moving it into a separate method makes things weirder because it makes the reader wonder what the importance is.

Reply:
My motivation is similar to the case with `copyCurrentReceipts`. This function is used in `commit` - the `for` loop is not necessary there (it is there only to create a log) and obfuscates the function, making slower to read. 

I general - I would be happy, and I hope others would also benefit, if we have more such abstractions.